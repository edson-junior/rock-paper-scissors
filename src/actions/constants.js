export const initialState = {
  user: 0,
  computer: 0,
  matches: 0,
  result: null,
};
export const ROCK = 'ROCK';
export const PAPER = 'PAPER';
export const SCISSORS = 'SCISSORS';
