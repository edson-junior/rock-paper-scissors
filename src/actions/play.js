import * as constants from './constants';

export const rock = rock => {
  type: constants.ROCK,
  rock
}

export const paper = paper => {
  type: constants.PAPER,
  paper
}

export const scissors = scissors => {
  type: constants.SCISSORS,
  scissors
}

