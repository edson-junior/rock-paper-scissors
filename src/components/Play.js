import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { addRock, addPaper, addScissors } from '../actions/play';

class Play extends Component {
  static propTypes = {
    onRock: PropTypes.func.isRequired,
    onPaper: PropTypes.func.isRequired,
    onScissors: PropTypes.func.isRequired,
  }
  render() {
    const { onRock, onPaper, onScissors } = this.props;

    return (
      <Fragment>
        <button className="rock" onClick={() => onRock('rock')}>rock</button>
        <button className="paper" onClick={() => onPaper('paper')}>paper</button>
        <button className="scissors" onClick={() => onScissors('scissors')}>scissors</button>
      </Fragment>
    );
  }
}

const mapStateToProps = state => ({
  store: state,
});

const mapDispatchToProps = dispatch => ({
  onRock: id => dispatch(addRock(id)),
  onPaper: id => dispatch(addPaper(id)),
  onScissors: id => dispatch(addScissors(id)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Play);
