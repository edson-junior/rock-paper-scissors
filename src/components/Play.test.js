import React from 'react';
import { mount, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Play from './Play';
import * as constants from '../actions/constants';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';

configure({ adapter: new Adapter() });

const initialState = constants.initialState;
const middlewares = []
const mockStore = configureStore(middlewares)
const store = mockStore(initialState);
const props = { store };
const play = mount(<Play { ...props } />);

describe('Play', () => {
  it('should render properly', () => {
    expect(toJson(play)).toMatchSnapshot();
  });

  it('should have a `Results` component', () => {
    expect(play.find('Results').exists()).toBe(true);
  });

  describe('when user clicks on an item', () => {
    let actions;

    beforeEach(() => actions = store.getActions());
    afterEach(() => store.clearActions());

    it('should dispatch rock', () => {
      const payload = { type: constants.ROCK, rock: 'rock' };

      play.find('.rock').simulate('click');
      expect(actions[0]).toEqual(payload);
    });

    it('should dispatch paper', () => {
      const payload = { type: constants.PAPER, paper: 'paper' };

      play.find('.paper').simulate('click');
      expect(actions[0]).toEqual(payload);
    });

    it('should dispatch scissors', () => {
      const payload = { type: constants.SCISSORS, scissors: 'scissors' };

      play.find('.scissors').simulate('click');
      expect(actions[0]).toEqual(payload);
    });

  });
});
