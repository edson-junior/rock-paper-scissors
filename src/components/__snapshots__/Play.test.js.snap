// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Play should render properly 1`] = `
<Connect(Play)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Play
    onPaper={[Function]}
    onRock={[Function]}
    onScissors={[Function]}
    store={Array []}
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <button
      className="rock"
      onClick={[Function]}
    >
      rock
    </button>
    <button
      className="paper"
      onClick={[Function]}
    >
      paper
    </button>
    <button
      className="scissors"
      onClick={[Function]}
    >
      scissors
    </button>
  </Play>
</Connect(Play)>
`;
